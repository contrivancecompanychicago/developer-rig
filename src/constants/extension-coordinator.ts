export enum ExtensionAnchor {
  Hidden = 'hidden',
  Panel = 'panel',
  VideoOverlay = 'videoOverlay',
  Component = 'component',
}

export enum ExtensionMode {
  Config = 'config',
  Dashboard = 'dashboard',
  Viewer = 'viewer',
}

export enum ExtensionPlatform {
  Mobile = 'mobile',
  Web = 'web',
}

export enum ExtensionState {
  Testing = 'Testing',
  HostedTest = 'Assets Uploaded',
  Approved = 'Approved',
  Released = 'Released',
  ReadyForReview = 'Ready For Review',
  InReview = 'In Review',
  PendingAction = 'Pending Action',
  Uploading = 'Uploading',
}

export enum ExtensionAction {
  TwitchExtAuth = 'twitch-ext-auth',
  TwitchExtBootstrap = 'twitch-ext-bootstrap',
  TwitchExtContext = 'twitch-ext-context',
  TwitchExtError = 'twitch-ext-error',
  TwitchExtLoaded = 'twitch-ext-loaded',
  TwitchExtLongtask = 'twitch-ext-longtask',
  TwitchExtNetworkTiming = 'twitch-ext-network-timing',
  TwitchExtReload = 'twitch-ext-reload',
  TwitchExtUserAction = 'twitch-ext-user-action',
  TwitchExtConfirmationRequest = 'twitch-ext-confirmation-request',
  TwitchExtBeginPurchase = 'twitch-ext-begin-purchase',
  TwitchExtReloadEntitlements = 'twitch-ext-reload-entitlements',
  TwitchExtProductPrices = 'twitch-ext-product-prices',
  TwitchExtVisibilityChanged = 'twitch-ext-visibility-changed',
  TwitchExtBitsProducts = 'twitch-ext-bits-products',
  TwitchExtUseBits = 'twitch-ext-use-bits',
  TwitchExtBitsTransactionComplete = 'twitch-ext-bits-transaction-complete',
  TwitchExtBitsOnHover = 'twitch-ext-bits-on-hover',
  TwitchExtPubSubReceived = 'twitch-ext-pubsub-received',
  TwitchExtPositionChanged = 'twitch-ext-position-changed',
  TwitchExtPubsubMessage = 'twitch-ext-pubsub-message',
  TwitchExtPubsubBindFailure = 'twitch-ext-pubsub-bind-failure',
  TwitchExtPubsubListen = 'twitch-ext-pubsub-listen',
  TwitchExtPubsubUnlisten = 'twitch-ext-pubsub-unlisten',
}

export enum ExtensionViewType {
  Component = "component",
  Config = "config",
  Hidden = "hidden",
  LiveConfig = "liveConfig",
  Mobile = "mobile",
  Panel = "panel",
  VideoOverlay = "videoOverlay",
}
