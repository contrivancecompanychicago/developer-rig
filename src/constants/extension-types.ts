import { ExtensionAnchor, ExtensionMode, ExtensionPlatform } from './extension-coordinator';

export const ExtensionAnchors: { [key: string]: string } = {
  [ExtensionAnchor.VideoOverlay]: 'Overlay',
  [ExtensionAnchor.Panel]: 'Panel',
  [ExtensionAnchor.Component]: 'Component',
  [ExtensionMode.Config]: 'Configuration',
  [ExtensionMode.Dashboard]: 'Dashboard',
  [ExtensionPlatform.Mobile]: 'Mobile',
};

export const DefaultExtensionType = ExtensionAnchor.VideoOverlay;
