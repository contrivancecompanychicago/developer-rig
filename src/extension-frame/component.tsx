import * as React from 'react';
import './component.sass';
import { RigExtension } from '../core/models/rig';
import { ExtensionPlatform, ExtensionViewType } from '../constants/extension-coordinator';

const IFRAME_CLASS = 'extension-frame';
const EXTENSION_FRAME_INIT_ACTION = 'extension-frame-init';

interface ExtensionFrameProps {
  channelId: string;
  className: string;
  frameId: string;
  extension: RigExtension;
  type: string;
  mode: string;
  bindIframeToParent: (iframe: HTMLIFrameElement) => void;
}

type Props = ExtensionFrameProps;

export class ExtensionFrame extends React.Component<Props> {
  public iframe: HTMLIFrameElement;

  public componentDidMount() {
    if (this.iframe) {
      this.iframe.onload = this.extensionFrameInit;
    }
  }

  public render() {
    return (
      <iframe
        ref={this.bindIframeRef}
        src={process.env.PUBLIC_URL + '/extension-frame.html'}
        frameBorder={0}
        className={'rig-frame ' + IFRAME_CLASS}
        title={this.props.frameId} />
    );
  }

  private bindIframeRef = (iframe: HTMLIFrameElement) => {
    this.iframe = iframe;
    this.props.bindIframeToParent(iframe);
  }

  public extensionFrameInit = () => {
    const extension: any = {
      anchor: this.props.type,
      channelId: this.props.channelId,
      loginId: null,
      extension: this.props.extension,
      mode: this.props.mode,
      platform: (this.props.type === ExtensionViewType.Mobile) ? ExtensionPlatform.Mobile : ExtensionPlatform.Web,
      trackingProperties: {},
      iframeClassName: IFRAME_CLASS,
    }
    const data = {
      extension: extension,
      action: EXTENSION_FRAME_INIT_ACTION,
      frameId: this.props.frameId,
    }
    this.iframe.contentWindow.postMessage(data, '*');
  }
}
