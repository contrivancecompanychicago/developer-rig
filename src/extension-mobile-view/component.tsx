import * as React from 'react';
import { ExtensionFrame } from '../extension-frame';
import { MobileOrientation } from '../constants/mobile';
import { RigExtension, FrameSize } from '../core/models/rig';
import { Position } from '../types/extension-coordinator';
import { ExtensionMode, ExtensionViewType} from '../constants/extension-coordinator';

const ViewBackgroundColor = '#322F37';
const AbsolutePosition = 'absolute';

interface ExtensionMobileViewProps {
  id: string;
  channelId: string;
  orientation: string;
  extension: RigExtension;
  frameSize: FrameSize;
  position: Position
  role: string;
  bindIframeToParent: (iframe: HTMLIFrameElement) => void;
}
type Props = ExtensionMobileViewProps & React.HTMLAttributes<HTMLDivElement>;

export class ExtensionMobileView extends React.Component<Props> {
  public computeFrameStyles(): React.CSSProperties {
    let frameStyles: React.CSSProperties;

    if (this.props.orientation === MobileOrientation.Portrait) {
      const height = Math.floor(this.props.frameSize.height * 0.65);
      frameStyles = {
        width: `${this.props.frameSize.width}px`,
        height: `${height}px`,
        bottom: '0'
      }
    } else {
      const width = Math.floor(this.props.frameSize.height * 0.28);
      frameStyles = {
        width: `${width}px`,
        height: `${this.props.frameSize.width}px`,
        right: '0'
      }
    }

    frameStyles.position = AbsolutePosition;
    return frameStyles;
  }

  public computeViewStyles() {
    let viewStyles: React.CSSProperties;
    if (this.props.orientation === MobileOrientation.Portrait) {
      viewStyles = {
        width: this.props.frameSize.width + 'px',
        height: this.props.frameSize.height + 'px',
      }
    } else {
      viewStyles = {
        width: this.props.frameSize.height + 'px',
        height: this.props.frameSize.width + 'px',
      }
    }

    viewStyles.background = ViewBackgroundColor;
    return viewStyles;
  }
  public render() {
    return (
      <div
        className="view component-view"
        style={this.computeViewStyles()}>
          <div style={this.computeFrameStyles()}>
          <ExtensionFrame
            bindIframeToParent={this.props.bindIframeToParent}
            className="view"
            channelId={this.props.channelId}
            frameId={`frameid-${this.props.id}`}
            extension={this.props.extension}
            type={ExtensionViewType.Mobile}
            mode={ExtensionMode.Viewer}
          />
        </div>
      </div>
    );
  }
}
