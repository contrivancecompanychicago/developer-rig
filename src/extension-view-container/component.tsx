import * as React from 'react';
import { ExtensionView } from '../extension-view';
import { ExtensionViewButton } from '../extension-view-button';
import { ExtensionMode } from '../constants/extension-coordinator';
import './component.sass';
import { RigExtensionView, RigExtension } from '../core/models/rig';
import { ConfigNames } from '../constants/nav-items'

interface ExtensionViewContainerProps {
  extensionViews: RigExtensionView[];
  openEditViewHandler?: (id: string) => void;
  deleteExtensionViewHandler: (id: string) => void;
  openExtensionViewHandler: Function;
}

type Props = ExtensionViewContainerProps;

export class ExtensionViewContainer extends React.Component<Props> {
  private openExtensionViewDialog = () => {
    this.props.openExtensionViewHandler();
  }

  public render() {
    let extensionViews: JSX.Element[] = [];
    if (this.props.extensionViews && this.props.extensionViews.length > 0) {
      extensionViews = this.props.extensionViews.map(view => {
        const role = view.mode === ExtensionMode.Viewer ? view.role : ConfigNames[view.mode];
        return <ExtensionView
          key={view.id}
          id={view.id}
          channelId={view.extension.channelId}
          extension={view.extension}
          type={view.type}
          mode={view.mode}
          role={role}
          frameSize={view.frameSize}
          position={{ x: view.x, y: view.y }}
          linked={view.linked}
          orientation={view.orientation}
          openEditViewHandler={this.props.openEditViewHandler}
          deleteViewHandler={this.props.deleteExtensionViewHandler} />
      });
    }

    return (
      <div className='view-container-wrapper'>
        <div className="view-container">
          {extensionViews}
        </div>
        <div>
          <ExtensionViewButton
            onClick={this.openExtensionViewDialog}>
          </ExtensionViewButton>
        </div>
      </div>
    );
  }
}
