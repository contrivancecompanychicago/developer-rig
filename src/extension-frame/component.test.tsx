import { setupShallowTest } from '../tests/enzyme-util/shallow';
import { setupMountTest } from '../tests/enzyme-util/mount';
import { ExtensionForTest } from '../tests/constants/extension';
import { ExtensionFrame } from './component';
import { ExtensionViewType, ExtensionAnchor, ExtensionMode, ExtensionState } from '../constants/extension-coordinator';

describe('<ExtensionFrame />', () => {
  const setupShallow = setupShallowTest(ExtensionFrame, () => ({
    frameId: '0',
    extension: ExtensionForTest,
    type: ExtensionAnchor.Panel,
    mode: ExtensionMode.Viewer,
    iframe: '',
    bindIframeToParent: jest.fn(),
  }));

  const setupMount = setupMountTest(ExtensionFrame, () => ({
    className: 'view',
    frameId: '0',
    extension: ExtensionForTest,
    type: ExtensionAnchor.Panel,
    mode: ExtensionMode.Viewer,
    iframe: '',
    bindIframeToParent: jest.fn(),
  }));

  it('onload postMessages data correctly', () => {
    const { wrapper } = setupMount();

    const mockIframeRef: any = {
      contentWindow: {
        postMessage: jest.fn(),
      },
    };

    const instance = wrapper.instance() as ExtensionFrame;
    instance.iframe = mockIframeRef;
    instance.extensionFrameInit();
    expect(mockIframeRef.contentWindow.postMessage).toHaveBeenCalledWith({
      'action': 'extension-frame-init',
      'extension': {
        'anchor': 'panel',
        'channelId': 'channelId',
        'extension': {
          'authorName': 'test',
          'bitsEnabled': false,
          'channelId': 'channelId',
          'description': 'description',
          'iconUrl': 'iconUrl',
          'id': 'id',
          'name': 'name',
          'requestIdentityLink': false,
          'sku': 'sku',
          'state': ExtensionState.Testing,
          'summary': 'summary',
          'token': 'token',
          'vendorCode': 'vendorCode',
          'version': '0.1',
          'views': {
            'component': { 'aspectHeight': 3000, 'aspectWidth': 2500, 'size': 0, 'viewerUrl': 'test', 'zoom': false, zoomPixels: 1024 },
            'config': { 'viewerUrl': 'test' },
            'liveConfig': { 'viewerUrl': 'test' },
            'videoOverlay': { 'viewerUrl': 'test' },
            'mobile': { 'viewerUrl': 'test' },
            'panel': { height: 300, 'viewerUrl': 'test' }
          },
          'whitelistedConfigUrls': ['foo'],
          'whitelistedPanelUrls': ['bar'] },
          'iframeClassName': 'extension-frame',
          'loginId': null,
          'mode': 'viewer',
          'platform': 'web',
          'trackingProperties': {}
        },
      'frameId': '0'
    }, '*');
  });

  it('onload postMessages data correctly when platform is mobile', () => {
    const { wrapper } = setupMount({
      type: ExtensionViewType.Mobile
    });

    const mockIframeRef: any = {
      contentWindow: {
        postMessage: jest.fn(),
      },
    };
    const instance = wrapper.instance() as ExtensionFrame;
    instance.iframe = mockIframeRef;
    instance.extensionFrameInit();
    expect(mockIframeRef.contentWindow.postMessage).toHaveBeenCalledWith({
      'action': 'extension-frame-init',
      'extension': {
        'anchor': 'mobile',
        'channelId': 'channelId',
        'extension': {
          'authorName': 'test',
          'bitsEnabled': false,
          'channelId': 'channelId',
          'description': 'description',
          'iconUrl': 'iconUrl',
          'id': 'id',
          'name': 'name',
          'requestIdentityLink': false,
          'sku': 'sku',
          'state': ExtensionState.Testing,
          'summary': 'summary',
          'token': 'token',
          'vendorCode': 'vendorCode',
          'version': '0.1',
          'views': {
            'component': { 'aspectHeight': 3000, 'aspectWidth': 2500, 'size': 0, 'viewerUrl': 'test', 'zoom': false, zoomPixels: 1024 },
            'config': { 'viewerUrl': 'test' },
            'liveConfig': { 'viewerUrl': 'test' },
            'videoOverlay': { 'viewerUrl': 'test' },
            'mobile': { 'viewerUrl': 'test' },
            'panel': { height: 300, 'viewerUrl': 'test' }
          },
          'whitelistedConfigUrls': ['foo'],
          'whitelistedPanelUrls': ['bar'] },
          'iframeClassName': 'extension-frame',
          'loginId': null,
          'mode': 'viewer',
          'platform': 'mobile',
          'trackingProperties': {}
        },
      'frameId': '0'
    }, '*');
  });

  describe('when in live config mode', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow({
        type: ExtensionViewType.LiveConfig,
        mode: ExtensionMode.Dashboard,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when in live config mode', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow({
        type: ExtensionViewType.LiveConfig,
        mode: ExtensionMode.Dashboard,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when in config mode', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow({
        type: ExtensionViewType.Config,
        mode: ExtensionMode.Config,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when in panel mode', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow({
        type: ExtensionAnchor.Panel,
        mode: ExtensionMode.Viewer,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when in video overlay mode', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow({
        type: ExtensionAnchor.VideoOverlay,
        mode: ExtensionMode.Viewer,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when in video overlay mode', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow({
        type: ExtensionAnchor.VideoOverlay,
        mode: ExtensionMode.Viewer,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });
});
