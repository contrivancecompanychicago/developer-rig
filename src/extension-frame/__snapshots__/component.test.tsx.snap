// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExtensionFrame /> when in config mode renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionFrame
    bindIframeToParent={[Function]}
    extension={
        Object {
            "anchor": "panel",
            "authorName": "test",
            "bitsEnabled": false,
            "canInstall": true,
            "channelId": "channelId",
            "description": "description",
            "eulaTosUrl": "test.biz",
            "iconUrl": "iconUrl",
            "iconUrls": Object {},
            "id": "id",
            "installationCount": 0,
            "name": "name",
            "panelHeight": 300,
            "privacyPolicyUrl": "test.com",
            "requestIdentityLink": false,
            "requiredBroadcasterAbilities": Array [
              "test",
            ],
            "screenshotUrls": Array [
              "test.png",
            ],
            "sku": "sku",
            "state": "Testing",
            "summary": "summary",
            "supportEmail": "test",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "size": 0,
                "viewerUrl": "test",
                "zoom": false,
                "zoomPixels": 1024,
              },
              "config": Object {
                "viewerUrl": "test",
              },
              "liveConfig": Object {
                "viewerUrl": "test",
              },
              "mobile": Object {
                "viewerUrl": "test",
              },
              "panel": Object {
                "height": 300,
                "viewerUrl": "test",
              },
              "videoOverlay": Object {
                "viewerUrl": "test",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    frameId="0"
    iframe=""
    mode="config"
    type="config"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "rig-frame extension-frame",
      "frameBorder": 0,
      "src": "/extension-frame.html",
      "title": "0",
    },
    "ref": [Function],
    "rendered": null,
    "type": "iframe",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "rig-frame extension-frame",
        "frameBorder": 0,
        "src": "/extension-frame.html",
        "title": "0",
      },
      "ref": [Function],
      "rendered": null,
      "type": "iframe",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ExtensionFrame /> when in live config mode renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionFrame
    bindIframeToParent={[Function]}
    extension={
        Object {
            "anchor": "panel",
            "authorName": "test",
            "bitsEnabled": false,
            "canInstall": true,
            "channelId": "channelId",
            "description": "description",
            "eulaTosUrl": "test.biz",
            "iconUrl": "iconUrl",
            "iconUrls": Object {},
            "id": "id",
            "installationCount": 0,
            "name": "name",
            "panelHeight": 300,
            "privacyPolicyUrl": "test.com",
            "requestIdentityLink": false,
            "requiredBroadcasterAbilities": Array [
              "test",
            ],
            "screenshotUrls": Array [
              "test.png",
            ],
            "sku": "sku",
            "state": "Testing",
            "summary": "summary",
            "supportEmail": "test",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "size": 0,
                "viewerUrl": "test",
                "zoom": false,
                "zoomPixels": 1024,
              },
              "config": Object {
                "viewerUrl": "test",
              },
              "liveConfig": Object {
                "viewerUrl": "test",
              },
              "mobile": Object {
                "viewerUrl": "test",
              },
              "panel": Object {
                "height": 300,
                "viewerUrl": "test",
              },
              "videoOverlay": Object {
                "viewerUrl": "test",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    frameId="0"
    iframe=""
    mode="dashboard"
    type="liveConfig"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "rig-frame extension-frame",
      "frameBorder": 0,
      "src": "/extension-frame.html",
      "title": "0",
    },
    "ref": [Function],
    "rendered": null,
    "type": "iframe",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "rig-frame extension-frame",
        "frameBorder": 0,
        "src": "/extension-frame.html",
        "title": "0",
      },
      "ref": [Function],
      "rendered": null,
      "type": "iframe",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ExtensionFrame /> when in live config mode renders correctly 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionFrame
    bindIframeToParent={[Function]}
    extension={
        Object {
            "anchor": "panel",
            "authorName": "test",
            "bitsEnabled": false,
            "canInstall": true,
            "channelId": "channelId",
            "description": "description",
            "eulaTosUrl": "test.biz",
            "iconUrl": "iconUrl",
            "iconUrls": Object {},
            "id": "id",
            "installationCount": 0,
            "name": "name",
            "panelHeight": 300,
            "privacyPolicyUrl": "test.com",
            "requestIdentityLink": false,
            "requiredBroadcasterAbilities": Array [
              "test",
            ],
            "screenshotUrls": Array [
              "test.png",
            ],
            "sku": "sku",
            "state": "Testing",
            "summary": "summary",
            "supportEmail": "test",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "size": 0,
                "viewerUrl": "test",
                "zoom": false,
                "zoomPixels": 1024,
              },
              "config": Object {
                "viewerUrl": "test",
              },
              "liveConfig": Object {
                "viewerUrl": "test",
              },
              "mobile": Object {
                "viewerUrl": "test",
              },
              "panel": Object {
                "height": 300,
                "viewerUrl": "test",
              },
              "videoOverlay": Object {
                "viewerUrl": "test",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    frameId="0"
    iframe=""
    mode="dashboard"
    type="liveConfig"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "rig-frame extension-frame",
      "frameBorder": 0,
      "src": "/extension-frame.html",
      "title": "0",
    },
    "ref": [Function],
    "rendered": null,
    "type": "iframe",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "rig-frame extension-frame",
        "frameBorder": 0,
        "src": "/extension-frame.html",
        "title": "0",
      },
      "ref": [Function],
      "rendered": null,
      "type": "iframe",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ExtensionFrame /> when in panel mode renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionFrame
    bindIframeToParent={[Function]}
    extension={
        Object {
            "anchor": "panel",
            "authorName": "test",
            "bitsEnabled": false,
            "canInstall": true,
            "channelId": "channelId",
            "description": "description",
            "eulaTosUrl": "test.biz",
            "iconUrl": "iconUrl",
            "iconUrls": Object {},
            "id": "id",
            "installationCount": 0,
            "name": "name",
            "panelHeight": 300,
            "privacyPolicyUrl": "test.com",
            "requestIdentityLink": false,
            "requiredBroadcasterAbilities": Array [
              "test",
            ],
            "screenshotUrls": Array [
              "test.png",
            ],
            "sku": "sku",
            "state": "Testing",
            "summary": "summary",
            "supportEmail": "test",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "size": 0,
                "viewerUrl": "test",
                "zoom": false,
                "zoomPixels": 1024,
              },
              "config": Object {
                "viewerUrl": "test",
              },
              "liveConfig": Object {
                "viewerUrl": "test",
              },
              "mobile": Object {
                "viewerUrl": "test",
              },
              "panel": Object {
                "height": 300,
                "viewerUrl": "test",
              },
              "videoOverlay": Object {
                "viewerUrl": "test",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    frameId="0"
    iframe=""
    mode="viewer"
    type="panel"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "rig-frame extension-frame",
      "frameBorder": 0,
      "src": "/extension-frame.html",
      "title": "0",
    },
    "ref": [Function],
    "rendered": null,
    "type": "iframe",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "rig-frame extension-frame",
        "frameBorder": 0,
        "src": "/extension-frame.html",
        "title": "0",
      },
      "ref": [Function],
      "rendered": null,
      "type": "iframe",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ExtensionFrame /> when in video overlay mode renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionFrame
    bindIframeToParent={[Function]}
    extension={
        Object {
            "anchor": "panel",
            "authorName": "test",
            "bitsEnabled": false,
            "canInstall": true,
            "channelId": "channelId",
            "description": "description",
            "eulaTosUrl": "test.biz",
            "iconUrl": "iconUrl",
            "iconUrls": Object {},
            "id": "id",
            "installationCount": 0,
            "name": "name",
            "panelHeight": 300,
            "privacyPolicyUrl": "test.com",
            "requestIdentityLink": false,
            "requiredBroadcasterAbilities": Array [
              "test",
            ],
            "screenshotUrls": Array [
              "test.png",
            ],
            "sku": "sku",
            "state": "Testing",
            "summary": "summary",
            "supportEmail": "test",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "size": 0,
                "viewerUrl": "test",
                "zoom": false,
                "zoomPixels": 1024,
              },
              "config": Object {
                "viewerUrl": "test",
              },
              "liveConfig": Object {
                "viewerUrl": "test",
              },
              "mobile": Object {
                "viewerUrl": "test",
              },
              "panel": Object {
                "height": 300,
                "viewerUrl": "test",
              },
              "videoOverlay": Object {
                "viewerUrl": "test",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    frameId="0"
    iframe=""
    mode="viewer"
    type="videoOverlay"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "rig-frame extension-frame",
      "frameBorder": 0,
      "src": "/extension-frame.html",
      "title": "0",
    },
    "ref": [Function],
    "rendered": null,
    "type": "iframe",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "rig-frame extension-frame",
        "frameBorder": 0,
        "src": "/extension-frame.html",
        "title": "0",
      },
      "ref": [Function],
      "rendered": null,
      "type": "iframe",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ExtensionFrame /> when in video overlay mode renders correctly 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionFrame
    bindIframeToParent={[Function]}
    extension={
        Object {
            "anchor": "panel",
            "authorName": "test",
            "bitsEnabled": false,
            "canInstall": true,
            "channelId": "channelId",
            "description": "description",
            "eulaTosUrl": "test.biz",
            "iconUrl": "iconUrl",
            "iconUrls": Object {},
            "id": "id",
            "installationCount": 0,
            "name": "name",
            "panelHeight": 300,
            "privacyPolicyUrl": "test.com",
            "requestIdentityLink": false,
            "requiredBroadcasterAbilities": Array [
              "test",
            ],
            "screenshotUrls": Array [
              "test.png",
            ],
            "sku": "sku",
            "state": "Testing",
            "summary": "summary",
            "supportEmail": "test",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "size": 0,
                "viewerUrl": "test",
                "zoom": false,
                "zoomPixels": 1024,
              },
              "config": Object {
                "viewerUrl": "test",
              },
              "liveConfig": Object {
                "viewerUrl": "test",
              },
              "mobile": Object {
                "viewerUrl": "test",
              },
              "panel": Object {
                "height": 300,
                "viewerUrl": "test",
              },
              "videoOverlay": Object {
                "viewerUrl": "test",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    frameId="0"
    iframe=""
    mode="viewer"
    type="videoOverlay"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "rig-frame extension-frame",
      "frameBorder": 0,
      "src": "/extension-frame.html",
      "title": "0",
    },
    "ref": [Function],
    "rendered": null,
    "type": "iframe",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "rig-frame extension-frame",
        "frameBorder": 0,
        "src": "/extension-frame.html",
        "title": "0",
      },
      "ref": [Function],
      "rendered": null,
      "type": "iframe",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
